internal/connector/internal/handlers/connector_agent.go

```diff
@@ -222,9 +222,34 @@ func (h *ConnectorClusterHandler) presentDeployment(r *http.Request, resource db
 		return private.ConnectorDeployment{}, err
 	}
 
-	apiSpec, err := h.Service.GetConnectorWithBase64Secrets(r.Context(), resource)
+	connectorPhase := resource.Status.Phase
+	// try resetting phase if secrets were previously missing
+	if connectorPhase == dbapi.ConnectorStatusPhaseSecretsUnavailable {
+		connectorPhase = dbapi.ConnectorStatusPhaseAssigned
+	}
+
+	// avoid ignoring this deployment altogether if there is an issue in getting secrets from the vault
+	ctx := r.Context()
+	apiSpec, invalidSecrets, err := h.Service.GetConnectorWithBase64Secrets(ctx, resource)
 	if err != nil {
-		return private.ConnectorDeployment{}, err
+		if invalidSecrets {
+			// log error in getting secrets and signal that connector spec doesn't have secrets
+			glog.Errorf("Error getting connector %s with base64 secrets: %s", apiSpec.ID, err)
+
+			apiSpec.ConnectorSpec = []byte("{}")
+			connectorPhase = dbapi.ConnectorStatusPhaseSecretsUnavailable
```

Comment:
this mean that the "missing-secrets" is ephemeral right ? si it is not persisted in the database

Reply:
See line 250 where it's persisted to DB.

Reply:
Since the status is owned by the fleet shard, would this be overridden by a status update from the sync ? 
It feel wrong to me that the fleet manager manipulate the deployment status.

Reply:
Yeah, I agree. It's violating the producer/consumer roles. I'm going to remove that.

Reply:
Removed fleetmanager DB update.